### DISCUSSION TITLE
Restrict API calls using API Gateway

### DISCUSSION THREAD

**Discussion by CROSS0822** [cloud-custodian/cloud-custodian#7768] [AWS]
I am looking for help in developing a couple/few Cloud Custodian policies for the following: 

1. How to restrict/prevent API calls unless the conditions match: "apigateway:Request/EndpointType": "PRIVATE" and "apigateway:Resource/EndpointType": "PRIVATE"
2. How to detect/notify on any API calls made where: 
- [ ] "apigateway:Request/EndpointType": "PUBLIC" and "apigateway:Resource/EndpointType": "PUBLIC"
- [ ] "apigateway:Request/EndpointType": "PUBLIC" and "apigateway:Resource/EndpointType": "EDGE"

I appreciate any assistance in putting these together, or input that the community has!

**Comment by MajinBuuOnSecurity** [comment]
https://docs.aws.amazon.com/apigateway/latest/developerguide/security_iam_id-based-policy-examples.html#security_iam_id-based-policy-examples-private-api
